azure-iot-device/tests/common/test_connection_string.py

```diff
@@ -8,61 +8,68 @@
 from azure.iot.device.common.connection_string import ConnectionString
 
 
+@pytest.mark.describe("ConnectionString")
 class TestConnectionString(object):
+    @pytest.mark.it("Instantiates from a given connection string")
     @pytest.mark.parametrize(
         "input_string",
         [
             pytest.param(
                 "HostName=my.host.name;SharedAccessKeyName=mykeyname;SharedAccessKey=Zm9vYmFy",
-                id="service string",
+                id="Service connection string",
             ),
             pytest.param(
                 "HostName=my.host.name;DeviceId=my-device;SharedAccessKey=Zm9vYmFy",
-                id="device string",
+                id="Device connection string",
             ),
             pytest.param(
                 "HostName=my.host.name;DeviceId=my-device;SharedAccessKey=Zm9vYmFy;GatewayHostName=mygateway",
-                id="device string with gatewayhostname",
+                id="Device connection string w/ gatewayhostname",
             ),
             pytest.param(
                 "HostName=my.host.name;DeviceId=my-device;ModuleId=my-module;SharedAccessKey=Zm9vYmFy",
-                id="module string",
+                id="Module connection string",
             ),
             pytest.param(
                 "HostName=my.host.name;DeviceId=my-device;ModuleId=my-module;SharedAccessKey=Zm9vYmFy;GatewayHostName=mygateway",
-                id="module string with gatewayhostname",
+                id="Module connection string w/ gatewayhostname",
             ),
         ],
     )
     def test_instantiates_correctly_from_string(self, input_string):
         cs = ConnectionString(input_string)
         assert isinstance(cs, ConnectionString)
 
+    @pytest.mark.it("Raises ValueError on bad input")
     @pytest.mark.parametrize(
         "input_string",
         [
-            pytest.param("", id="empty string"),
-            pytest.param("garbage", id="garbage"),
-            pytest.param("HostName=my.host.name", id="incomplete connection string"),
+            pytest.param("", id="Empty string"),
+            pytest.param("garbage", id="Not a connection string"),
+            pytest.param("HostName=my.host.name", id="Incomplete connection string"),
             pytest.param(
                 "InvalidKey=my.host.name;SharedAccessKeyName=mykeyname;SharedAccessKey=Zm9vYmFy",
-                id="invalid key",
+                id="Invalid key",
             ),
             pytest.param(
                 "HostName=my.host.name;HostName=my.host.name;SharedAccessKey=mykeyname;SharedAccessKey=Zm9vYmFy",
-                id="duplicate key",
+                id="Duplicate key",
             ),
         ],
     )
     def test_raises_value_error_on_bad_input(self, input_string):
         with pytest.raises(ValueError):
             ConnectionString(input_string)
 
+    @pytest.mark.it(
+        "Provides the input connection string as a string representation for an instance"
```

Comment:
>Provides the input connection string as a string representation for an instance" [](start = 9, length = 80)

Awkward.   Maybe "Returns the input connection string as its string representation"?

Reply:
Another suggestion combining both :

Returns the input connection string as the string representation of an instance

---
In reply to: [290982473](https://github.com/Azure/azure-iot-sdk-python-preview/pull/101#discussion_r290982473) [](ancestors = 290982473)

Reply:
Since the word "return" has certain implications, I decided to go with

"Uses the input connection string as a string representation"

I don't need to specify "for an instance", I think that's implied

---
In reply to: [290988868](https://github.com/Azure/azure-iot-sdk-python-preview/pull/101#discussion_r290988868) [](ancestors = 290988868,290982473)